src/util/alloc/allocator.rs

```diff
@@ -66,7 +66,7 @@ pub fn align_allocation_inner<VM: VMBinding>(
 
     let mask = (alignment - 1) as isize; // fromIntSignExtend
     let neg_off = -offset; // fromIntSignExtend
-    let delta = (neg_off - region_isize) & mask;
+    let delta = neg_off.wrapping_sub(region_isize) & mask; // Use wrapping_sub to avoid overflow
```

Comment:
Since Rust 1.66.0 (our rust-toolchain is 1.66.1), we can use `isize::wrapping_sub_usize` and eliminate the `region_isize` variable.

```rust
    let delta = neg_off.wrapping_sub_usize(region) & mask; // Use wrapping subtraction to avoid overflow
```

Alternatively, we can use `usize::wrapped_add_signed` to eliminate all variables of the `isize` type.

```rust
    let mask = alignment - 1;
    let delta = region.wrapping_neg().wrapping_add_signed(neg_off) & mask; // Use wrapping operations to avoid overflow
```

or

```rust
    let mask = alignment - 1;
    let delta = region.wrapping_add_signed(offset).wrapping_neg() & mask; // Use wrapping operations to avoid overflow
```

I am not sure if any of them is a good idea because none of them is significantly more readable than others.  Mathematically, they are just various ways to express that "`delta ≡ - region - offset` (mod `alignment`)"

Reply:
This is a good idea. I gave it a try. But this requires us to bump our minimal support Rust version to 1.66. I feel it is not worth bumping MSRV merely for this change. I added a TODO for the line, and still kept the original code.